### PR TITLE
Don't instantiate HtmlCodec in CanvasKit mode

### PR DESCRIPTION
**Pull request by hterkelsen**
Fixes https://github.com/flutter/flutter/issues/55476

This is a temporary fix which gives a sensible error message, and also
prevents HTML-backend classes from leaking into CanvasKit.

### CODE REVIEW

File: lib/web_ui/lib/src/engine/compositor/canvas.dart [hunk 1]
```diff
@@ -122,6 +122,8 @@ class SkCanvas {
   }
 
   void drawImageRect(ui.Image image, ui.Rect src, ui.Rect dst, SkPaint paint) {
+    print("Drawing Image : " + image.toString());
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/compositor/canvas.dart:125]: remove print

File: lib/web_ui/lib/src/engine/compositor/canvas.dart [hunk 2]
```diff
@@ -122,6 +122,8 @@ class SkCanvas {
...
+    print(image.runtimeType);
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/compositor/canvas.dart:126]: remove

File: lib/web_ui/lib/src/engine/compositor/canvas.dart (hunk 1 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/canvas.dart]: Done.

File: lib/web_ui/lib/src/engine/compositor/canvas.dart (hunk 2 above)
Comment (hterkelsen) [reply lib/web_ui/lib/src/engine/compositor/canvas.dart]: Done.

**Comment by cbracken** [comment]
@hterkelsen looks like this is approved but has merge conflicts. Are you planning to pick this up again or shall we close it out for now and submit a new PR when it's good to go?

**Comment by cbracken** [comment]
Actually just took a look at the current code. Looks like this is obsolete. Please re-open if I'm wrong :)

**Pull request closed without merging**